chore: add about section

pratikmdhr opened the pull request:
## WHAT IT DOES:

_Description of the pull request, what changed_

-Added the about section to the homepage
-Will need to scroll down beyond viewport height to reach the About section

pratikmdhr committed:
chore: add about page (only displaying it on homepage)

pratikmdhr committed:
fix: check if valid json before parsing

pratikmdhr committed:
Merge branch 'main' into pratikmdhr/about-page

MaiaS commented on src/app/clientPage.tsx:
```diff
@@ -93,7 +98,8 @@ export default function ClientPage({ searchParams }: { searchParams?: { [key: st
   return (
-    <>
+    // Subtracting footer height from 100vh
+    <div className="min-h-[calc(100vh-7.9rem)] md:min-h-[calc(100vh-5rem)]">
```
Perhaps we want to keep a reference to the height in a variable so we can reuse?

MaiaS commented on src/app/clientPage.tsx:
```diff
@@ -93,7 +98,8 @@ export default function ClientPage({ searchParams }: { searchParams?: { [key: st
   return (
-    <>
+    // Subtracting footer height from 100vh
+    <div className="min-h-[calc(100vh-7.9rem)] md:min-h-[calc(100vh-5rem)]">
```
Perhaps we want to keep a reference to the height in a variable so we can reuse? Or in another container component?

pratikmdhr replied:
good idea!! i think storing as a variable would be better..that way we can use the same value in the footer component too

MaiaS commented on src/app/page.tsx:
```diff
@@ -1,8 +1,16 @@
+    <>
+      <ClientPage searchParams={searchParams} />
+      {/* Hide about page in the results page */}
+      {!isValidUrlWithEncodedState(searchParams) && <About />} 
```
This check isn't working too well for me. Almost like the state is not reactive

pratikmdhr replied:
do you know if there is a better way to check if the parsed string will be a JSON? I think this test would fail if the query params value are numbers, something like this /?result=123&original=2323...

I have tested the following cases:

[FAIL:  it goes to the Result page but is blank // it should have stayed in the homepage with about section displayed]
https://deploy-preview-26--sum-it-up.netlify.app/?original=2323&result=123

[PASS: it stays on the homepage as intended and shows About section as intended]
https://deploy-preview-26--sum-it-up.netlify.app/?original=&result=  
https://deploy-preview-26--sum-it-up.netlify.app/?original=abc&result=abc

pratikmdhr replied:
I will do some more tests and fix this later.. also I think we could even show the About section in the Result page, (we wouldn't really need to check the state then 😂)

pratikmdhr replied:
Ok I see the state also getting messed up after returning to the main page from the results page, will fix this later

MaiaS replied:
Sorry I missed these comments last week!

MaiaS replied:
as for this question...
>do you know if there is a better way to check if the parsed string will be a JSON? I think this test would fail if the query params value are numbers, something like this /?result=123&original=2323...

Didn't even notice that, nice catch. Perhaps we could check if the result has properties.

MaiaS reviewed (changes requested):
There's a small bug with how the about page shows and hides :). But looks good thank you!

pratikmdhr committed:
Merge branch 'main' into pratikmdhr/about-page

pratikmdhr committed:
fix: add a new tailwind height property for footer height

pratikmdhr committed:
refactor: render about page directly in the ClientPage component

Even though we "use-client" for ClientPage component, it will still get pre-rendered in the server (https://beta.nextjs.org/docs/rendering/server-and-client-components#client-components)

pratikmdhr committed:
style: add some styling to about section

pratikmdhr commented on src/app/clientPage.tsx:
```diff
@@ -100,22 +103,27 @@ export default function ClientPage({
+              onLengthChange={trackLengthSelection}
+            />
+          </MinHeightBodyContainer>
+          <About />
```
[this will get pre-rendered from the server](https://beta.nextjs.org/docs/rendering/server-and-client-components#client-components)

MaiaS reviewed (approved):
Thank you for fixing the issue!

pratikmdhr committed:
Merge branch 'main' into pratikmdhr/about-page
